example/lib/bloc/airship_bloc.dart

```diff
@@ -0,0 +1,154 @@
+  StreamSink<bool> get notificationsEnabledSetSink => _notificationsEnabledSetStreamController.sink;
+
+  Stream<List<String>> get tagsStream => _tagsStreamController.stream;
+  StreamSink<List<String>> get tagsSink => _tagsStreamController.sink;
+
+  Stream<List<InboxMessage>> get messagesStream => _messagesStreamController.stream;
+  StreamSink<List<InboxMessage>> get messagesSink => _messagesStreamController.sink;
+
+  StreamSink<List<String>> get tagsAddedSink => _tagsAddedStreamController.sink;
+  StreamSink<List<String>> get tagsRemovedSink => _tagsRemovedStreamController.sink;
+
+  StreamSink<InboxMessage> get messageRemovedSink => _messageRemovedStreamController.sink;
+
+  AirshipBloc() {
+    Airship.channelId.then((val) {
+      _channelStreamController.add(val);
+    });
+
+    Airship.tags.then((val) {
+      _tagsStreamController.add(val);
+    });
+
+    _tagsAddedStreamController.stream.listen((tags) {
+      Airship.addTags(tags).then((val) {
+        // Add the tags then grab fresh tags list
+        Airship.tags.then((airshipTags) {
+          tagsSink.add(airshipTags);
+        });
+      });
+    });
+
+    _tagsRemovedStreamController.stream.listen((tags) {
+      Airship.removeTags(tags).then((val) {
+        // Remove the tags then grab fresh tags list
+        Airship.tags.then((airshipTags) {
+          tagsSink.add(airshipTags);
+        });
+      });
+    });
+
+    Airship.namedUser.then((val) {
+      _namedUserStreamController.add(val);
+    });
+
+    _namedUserSetStreamController.stream.listen((namedUser) {
+      Airship.setNamedUser(namedUser).then((val) {
+        namedUserSink.add(namedUser);
+      });
+    });
+
+    Airship.userNotificationsEnabled.then((val) {
+      _notificationsEnabledStreamController.add(val);
+    });
+
+    _notificationsEnabledSetStreamController.stream.listen((enabled) {
+      Airship.setUserNotificationsEnabled(enabled).then((val) {
+        notificationsEnabledSink.add(enabled);
+      });
+    });
+
+    Airship.inboxMessages.then((messages) {
+      _messagesStreamController.add(messages);
+    });
+
+    _messageAddedStreamController.stream.listen((messages) {
+      Airship.inboxMessages.then((messages) {
+        messagesSink.add(messages);
+      });
+    });
+
+    _messageRemovedStreamController.stream.listen((message) {
+      Airship.deleteInboxMessage(message).then((onValue) {
+        // Get fresh message list
+        Airship.inboxMessages.then((messages) {
+          messagesSink.add(messages);
+        });
+      });
+    });
+
+    Airship.onInboxUpdated.listen((event) {
+      debugPrint('Inbox updated link');
+      // Get fresh message list
+      Airship.inboxMessages.then((messages) {
+        messagesSink.add(messages);
+      });
+    });
+
+    Airship.onShowInboxMessage.listen((messageId) {
+      //MaterialPageRoute(builder: (context) => MessageView(message: message));
+      debugPrint('Show inbox message $messageId');
```

Comment:
Is the actual `show` implemented somewhere else?

Reply:
Same as below, this is going to require the work to support deep linking, which requires some research on my part - and will be best to review as a separate PR.